lib/ansible/modules/database/postgresql/postgresql_db.py

```diff
@@ -262,7 +262,28 @@ def db_dump(module, target, target_opts="",
         cmd += " {0} ".format(target_opts)
 
     if comp_prog_path:
-        cmd = '{0}|{1} > {2}'.format(cmd, comp_prog_path, pipes.quote(target))
+        env = os.environ.copy()
+        if password:
+            env["PGPASSWORD"] = password
+        p1 = subprocess.Popen(cmd,
```

Comment:
I am not sure why we are not using `module.run_command` instead of `Popen`

Reply:
I think I understand what you mean. I have reverted those changes.